src/test/java/org/springframework/data/elasticsearch/core/MappingBuilderTests.java

```diff
@@ -53,6 +55,13 @@
 	@Autowired
 	private ElasticsearchTemplate elasticsearchTemplate;
 
+	private String xContentBuilderToString(XContentBuilder builder)
+	{
+		builder.close();
+		ByteArrayOutputStream bos = (ByteArrayOutputStream) builder.getOutputStream();
```

Comment:
It is possible for other kids of OutputStream to be assigned to the builder. The default is ByteArrayOutputStream.  It is a tricky one since it is a test we control. I was thinking in case the default changed or picked up some configuration that changed it, the problem might be more obvious if it blew up on the cast rather than the String compare.